spec.md

```diff
@@ -841,11 +841,61 @@ For success responses, the following fields are defined:
 | operation | string | For asynchronous responses, Service Brokers MAY return an identifier representing the operation. The value of this field MUST be provided by the Platform with requests to the [Last Operation](#polling-last-operation) endpoint in a percent-encoded query parameter. If present, MUST be a non-empty string. |
+| extension_apis | array-of-objects | For extensions to the Service Broker API, Service Brokers MAY return one or more `extension_api` objects that describe additional API endpoints via an OpenAPI document. See [Extension API Object](#extension-api-object) for more information. |
+
+##### Extension API Object
+
+The `extension_api` object MAY be used to describe any additional endpoint
+to the Open Service Broker API. An example of this could be lifecycle
+management of a Service Instance, (e.g. "Day Two Operations"), like Backup,
+Restore, Stop, Start, Restart and Pause.
+
+The `extension_api` MUST include a URI to an OpenAPI 3.0+ document that the
+Platform can use to determine the new endpoint(s), parameter(s), authentication
+mechanism and server URL. The new APIs are extensions to the Open Service Broker
+API. As such they are indended to be invoked by the Platform on behalf of its
+clients.
+
+Extension API endpoints MAY be executed on a remote server, however the OpenAPI
+document MUST include a Server Object with an accurate `url` parameter and a
+`description` field labeled, "Service Broker Extensions Server". If no Server
+Object `description` field contains, "Service Broker Extensions Server", the 
+Platform MUST assume the extension API endpoints are to be invoked using the
+Service Broker host and port. The `url` for this Server Object MUST be an
+absolute URL.
+
+```
+{
+  "servers": [
+    {
+      "url": "https://extensions.servicebrokerexample.com",
+      "description": "Service Broker Extensions Server"
+    }
+  ]
+}
+```
+
+See [OpenAPI Server Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#serverObject) for more information.
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| discovery_url* | string | A URI pointing to a valid OpenAPI 3.0+ document describing the API extension(s) to the Open Service Broker API including server location, endpoints, parameters and any other detail the platform needs for invocation. The location of the API extension endpoint(s) can be local to the Service Broker or on a remote server. MUST be a valid URI. The returned OpenAPI document MUST be in json format. See the [OpenAPI Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md) for more information. |
```

Comment:
For `dashboard_url` we include:
> The URL MUST contain enough information for the dashboard to identify the resource being accessed

I think we need something in this PR that allows tells people that something needs to be unique in all of the URLs/data being passed around, otherwise how does the generic action endpoint know which instance is being operated on.

Reply:
I don't agree that we MUST do this, but I will bow to the group decision. An example I can think of is you want to return the same swagger for every service instance and pass the service instance details via a body parameter, (as defined by the swagger).

Reply:
We can do that, IF we also define how someone can programmatically know which parameter in the swagger represents the instance ID. W/o that I wouldn't know how to write code to do it.

So, note I said "URL/data" - meaning I'm ok with a statement about instance-unique URLs, or other text talking about how to know which param to set - either is fine - we just need to be clear about how to do it.

Reply:
I think the latest address your concerns.

Reply:
@rhodie27 thanks - need to think more about it, at at first glace yes I think it does.

if we have an aggregation service broker in the path between the platform and the real service broker, I'm assuming the aggregation SB might need to twiddle relative URLs provided so they're not relative to the aggregation SB but point to the real one. Is this something we should say in here or is it obvious and we can leave that as an exercise for the dev?

Reply:
I'd lean towards "up to the dev". I could see the aggregation SB actually hosting the openapi doc itself instead of passing the request through. And it could have any combination of local and remote endpoints.